### ISSUE TITLE
many records

### ISSUE THREAD

**Issue by mschulze0** [susom/mass-delete#2]
Hi,

the problem with this module is that it tries to list every record. We have so many records in one of our projects that the website never renders. Is there a way to use this module by just pasting in a list of records to be deleted?

**Comment by 123andy** [comment]
Anything is possible, but we have no plans to do this.  If you have a large number of records to be erased, you could always use a direct sql method, as in:

`delete from redcap_data where project_id = xxx and record like '100%;`
or
`delete from redcap_data where project_id = xxx and record in (a,b,c);`

**Comment by 123andy** [comment]
You might also try using a different browser, maybe brave or safari if a mac, or edge if a PC.   I'm guessing the holdup is in the rendering of the HTML for the selector box...

**Comment by mschulze0** [comment]
> Anything is possible, but we have no plans to do this. If you have a large number of records to be erased, you could always use a direct sql method, as in:
> 
> `delete from redcap_data where project_id = xxx and record like '100%;`
> or
> `delete from redcap_data where project_id = xxx and record in (a,b,c);`

yes, but that doesn't take care of deleting the log entries

**Comment by mschulze0** [comment]
> You might also try using a different browser, maybe brave or safari if a mac, or edge if a PC. I'm guessing the holdup is in the rendering of the HTML for the selector box...

indeed it is. that's why it would be nice to be able to turn that off and go straight to a text box that lets me paste the records. but also looking at the code it seems like the whole "enter a list of records" is dependent on there being checkboxes that then get checked for the records entered.

actually in the meantime I also tried the older plugin batch_delete_records which actually does offer a text box straight away. but even that completely hangs up the browser when pasting 30,000 IDs in there lol
I managed to do it in three chunks of about 10,000 IDs haha
